### PR TITLE
ElectroNP flowsheet report

### PR DESCRIPTION
**Pull request by adam-a-a**
## Summary/Motivation:
Collaborative branch for electroN-P flowsheet with AD

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py [hunk 1]
```diff
@@ -177,99 +180,229 @@ def build_flowsheet():
...
+        "X_I",
+        "X_H",
+        "X_PP",
+        "X_S",
+    ]
+
+    for i in skip_constraint:
+        m.fs.electroNP.solute_removal_equation[0.0, "Liq", i].deactivate()
+        m.fs.electroNP.solute_treated_equation[0.0, "Liq", i].deactivate()
+
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py:243]: @agarciadiego identified these as degenerate. @luohezhiming will run DegeneracyHunter to see if he can identify degenerate constraints and confirm. If confirmed, revise the electronP unit

**Comment by adam-a-a** [comment]
@luohezhiming @MarcusHolly  I think we can close this now since this was meant to help us streamline analysis for our initial report. Do you concur?

**Comment by MarcusHolly** [comment]
> @luohezhiming @MarcusHolly I think we can close this now since this was meant to help us streamline analysis for our initial report. Do you concur?

Are we sure there are no additional changes we actually want to keep? For instance, the change made to the watertap_costing package? I haven't gone through every change made in this PR, but it does look like most of the changes have already been merged into main by a different PR(s).

**Comment by luohezhiming** [comment]
> > @luohezhiming @MarcusHolly I think we can close this now since this was meant to help us streamline analysis for our initial report. Do you concur?
> 
> Are we sure there are no additional changes we actually want to keep? For instance, the change made to the watertap_costing package? I haven't gone through every change made in this PR, but it does look like most of the changes have already been merged into main by a different PR(s).

I think the only remaining thing are those monte carlo simulation files, do we want to include it in WaterTAP? If not, then we can close this PR.

**Comment by adam-a-a** [comment]
This PR was meant to be an experimental draft for us to collaboratively run analyses for an internal report back around June 2023. No need to keep this open; we can still revisit code from closed PRs if there's anything we want to revitalize.

**Pull request closed without merging**